ndcube/ndcube.py

```diff
@@ -366,6 +372,98 @@ def extra_coords(self):
                     "value": self._extra_coords_wcs_axis[key]["value"]}
         return result
 
+    def all_world_coords(self, axes=None):
+        """
+        Returns WCS coordinate values of all pixels for all axes.
+
+        Parameters
+        ----------
+        axes: `int` or `str`, or iterable of `int` or `str`
+            Axis number in numpy ordering or unique substring of world_axis_physical_types
+            of axes for which real world coordinates are desired.
+            Default=None implies all axes will be returned.
+
+        Returns
+        -------
+        axes_coords: `list` of `astropy.units.Quantity`
+            Real world coords for axes in order requested by user.
+
+        Example
+        -------
+        >>> NDCube.all_world_coords(('lat', 'lon')) # doctest: +SKIP
+        >>> NDCube.all_world_coords(2) # doctest: +SKIP
+
+        """
+        # Define the dimensions of the cube and the total number of axes.
+        cube_dimensions = np.array(self.dimensions.value, dtype=int)
+        n_dimensions = len(cube_dimensions)
+        world_axis_types = self.world_axis_physical_types
+        # Determine axis numbers of user supplied axes.
+        if axes is None:
+            int_axes = np.arange(n_dimensions)
+        else:
+            if isinstance(axes, int):
+                int_axes = np.array([axes])
+            elif isinstance(axes, str):
+                int_axes = np.array([
+                    utils.cube.get_axis_number_from_axis_name(axes, world_axis_types)])
+            else:
+                int_axes = []
+                for axis in axes:
+                    if isinstance(axis, int):
+                        int_axes.append(axis)
+                    elif isinstance(axis, str):
+                        int_axes.append(
+                            utils.cube.get_axis_number_from_axis_name(axis, world_axis_types))
+                int_axes = np.array(int_axes)
+        # Ensure user has not entered the same axis twice.
+        if len(np.unique(int_axes)) != len(int_axes):
+            raise ValueError("Same axis entered more than once.")
+        n_axes = len(int_axes)
+        axes_coords = [None] * n_axes
+        axes_translated = np.array([False] * n_axes)
```

Comment:
this should be `np.zeros_like(int_axes, dtype=bool)`

Reply:
Done